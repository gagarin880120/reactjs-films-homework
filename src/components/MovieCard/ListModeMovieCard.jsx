import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './MovieCard.module.scss';
import TrailerButtonRect from '../TrailerButtonRect/TrailerButtonRect';

export default function ListModeMovieCard({
  movie, onTrailerButtonClick, onLinkClick,
}) {
  return (
    <div
      className={styles.listWrapper}
    >
      <div
        className={styles.listImage}
        style={{
          backgroundImage: movie.poster_path
            ? `url(https://image.tmdb.org/t/p/w300/${movie.poster_path})`
            : null,
        }}
      />
      <div className={styles.listMovieInfo}>
        <Link
          to={`/movie/${movie.id}`}
          onClick={() => {
            onLinkClick(movie.id);
          }}
        >
          <p className={styles.movieTitle}>{movie.title}</p>
        </Link>
        <span className={styles.movieRating}>{movie.vote_average}</span>
        <p className={styles.movieGenre}>{movie.genres}</p>

        <div className={styles.overview}>{movie.overview}</div>
        <TrailerButtonRect
          onTrailerButtonClick={onTrailerButtonClick}
          id={movie.id}
        />
      </div>
    </div>
  );
}

ListModeMovieCard.propTypes = {
  onTrailerButtonClick: PropTypes.func,
  onLinkClick: PropTypes.func,
  movie: PropTypes.shape({
    poster_path: PropTypes.string,
    id: PropTypes.number,
    title: PropTypes.string,
    vote_average: PropTypes.number,
    genres: PropTypes.string,
    overview: PropTypes.string,
  }),
};

ListModeMovieCard.defaultProps = {
  onTrailerButtonClick: null,
  onLinkClick: null,
  movie: {
    poster_path: '',
    id: 0,
    title: '',
    vote_average: 0,
    genres: '',
    overview: '',
  },
};
