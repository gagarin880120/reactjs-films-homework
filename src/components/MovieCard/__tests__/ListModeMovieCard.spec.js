import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer, { act } from 'react-test-renderer';
import React from 'react';
import ListModeMovieCard from '../ListModeMovieCard';
import { BrowserRouter as Router } from 'react-router-dom';

describe('ListModeMovieCard component', () => {
  const movie = {
    title: 'Star Wars',
    genre_ids: [28],
    id: 11
  }

  const movieWithPoster = {
    title: 'Star Wars',
    genre_ids: [28],
    id: 11,
    poster_path: 'path'
  }

  const genres = [{id:28, name:'Adventure'}]

  test('should render without crashing', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<ListModeMovieCard movie={movie} genres={genres} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  test('when Link element is clicked should call onLinkClick', () => {
    const onLinkClick = jest.fn();
    const testRenderer = TestRenderer.create(
      <Router>
        <ListModeMovieCard movie={movieWithPoster} onLinkClick={onLinkClick} />
      </Router>
    );
    const link = testRenderer.root.findByProps({ to: '/movie/11' });

    act(() => {
      link.props.onClick();
    });

    expect(onLinkClick).toHaveBeenCalled();
  });
});
