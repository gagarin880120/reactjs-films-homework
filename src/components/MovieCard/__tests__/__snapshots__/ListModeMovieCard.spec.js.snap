// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListModeMovieCard component should render without crashing 1`] = `
<div
  className="listWrapper"
>
  <div
    className="listImage"
    style={
      Object {
        "backgroundImage": null,
      }
    }
  />
  <div
    className="listMovieInfo"
  >
    <Link
      onClick={[Function]}
      to="/movie/11"
    >
      <p
        className="movieTitle"
      >
        Star Wars
      </p>
    </Link>
    <span
      className="movieRating"
    />
    <p
      className="movieGenre"
    />
    <div
      className="overview"
    />
    <TrailerButtonRect
      id={11}
      onTrailerButtonClick={null}
    />
  </div>
</div>
`;
